src/components/[guild]/RolePlatforms/components/AddRoleRewardModal/components/DynamicSetup/DynamicRewardSetup.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  Box,
+  Circle,
+  FormLabel,
+  HStack,
+  Icon,
+  Stack,
+  Text,
+  useDisclosure,
+} from "@chakra-ui/react"
+import RequirementDisplayComponent from "components/[guild]/Requirements/components/RequirementDisplayComponent"
+import useRequirements from "components/[guild]/hooks/useRequirements"
+import AddCard from "components/common/AddCard"
+import Button from "components/common/Button"
+import Card from "components/common/Card"
+import { ArrowsClockwise, Lightning } from "phosphor-react"
+import { useFormContext, useWatch } from "react-hook-form"
+import ConversionInput from "../AddTokenPanel/components/ConversionInput"
+import BaseValueModal from "./BaseValueModal"
+
+const DynamicRewardSetup = ({
+  toImage,
+  roleId,
+  requirementFieldName,
+  multiplierFieldName,
+  shouldFloor,
+}: {
+  toImage: JSX.Element
+  fieldName?: string
+  roleId: number
+  requirementFieldName: string
+  multiplierFieldName: string
+  shouldFloor?: boolean
+}) => {
+  const { control, setValue } = useFormContext()
+
+  const requirementId = useWatch({ name: requirementFieldName, control })
+
+  const { data: requirements } = useRequirements(roleId)
+  const selectedRequirement = requirements?.find((req) => req.id === requirementId)
+
+  const multiplier = useWatch({ name: multiplierFieldName, control })
+
+  const fromImage = (
+    <Circle bgColor={"white"} size={5} overflow="hidden">
+      <Icon boxSize={3} as={Lightning} weight="fill" color="green.500" />
+    </Circle>
+  )
+
+  const { onOpen, onClose, isOpen } = useDisclosure()
+
+  return (
+    <>
+      <Stack gap={4}>
+        <Box>
+          <HStack justifyContent={"space-between"} w="full" mb="2">
+            <FormLabel mb="0">Base value</FormLabel>
```

Comment:
Would be great to add error handling here, so we show red FormErrorMessage if the user tries to continue without setting a requirement